lib/miq_expression/tag.rb

```diff
@@ -1,17 +1,27 @@
-class MiqExpression::Tag
-  def self.parse(tag)
-    klass, ns = tag.split(".")
-    klass, ns = "nil", klass if ns.nil? # support managed-label
-    ns = "/" + ns.split("-").join("/")
-    ns = ns.sub(/(\/user_tag\/)/, "/user/") # replace with correct namespace for user tags
-    new(klass.safe_constantize, ns)
-  end
+class MiqExpression::Tag < MiqExpression::Field
```

Comment:
I'm not sure about this change (inheriting from `Field`). A `Tag` is not a specialized kind of `Field` - it already has quite a large surface area, many methods don't make sense in the context of tags. Perhaps you could share the code that's needed by pulling out an abstract class?